File: [5 kyu] Maximum subarray sum/main.cpp
#include <vector>

int maxSequence(const std::vector<int> &arr) {
    int sum = 0;
    int max = 0;

    for (const auto &val : arr) {
        sum += val;

        if (sum < 0) {
            sum = 0;
        } else {
            max = std::max(max, sum);
        }
    }

    return max;
}

#include <vector>

int maxSequence(const std::vector<int> &arr) {
    int sum = 0;
    int max = 0;

    for (const auto &val : arr) {
        sum = (sum + val < 0) ? 0 : sum + val;
        max = (sum > max) ? sum : max;
    }

    return max;
}

File: [5 kyu] Maximum subarray sum/main.rs
fn max_sequence(seq: &[i32]) -> i32 {
    let mut sum = 0;
    let mut max = 0;

    for &val in seq.iter() {
        sum = if sum + val < 0 { 0 } else { sum + val };
        max = if sum > max { sum } else { max }
    }

    max
}

fn max_sequence(seq: &[i32]) -> i32 {
    let mut sum = 0;
    let mut max = 0;

    for &val in seq.iter() {
        sum = (sum + val).max(0);
        max = sum.max(max);
    }

    max
}
